include/libdnf/base/base.hpp

```diff
@@ -31,6 +31,20 @@ namespace libdnf {
 /// :class:`.Base` instances are stateful objects owning various data.
 class Base {
 public:
+    /// Loads main configuration from file defined by path.
+    void load_config_from_file(const std::string & path);
+
+    /// Loads main configuration from file defined by the current configuration.
+    void load_config_from_file();
+
+    /// Loads main configuration from files with ".conf" extension from directory defined by dir_path.
+    /// The files in the directory are read in alphabetical order.
+    void load_config_from_dir(const std::string & dir_path);
+
+    /// Loads main configuration from files with ".conf" extension from directory defined by the current configuration.
+    /// The files in the directory are read in alphabetical order.
+    void load_config_from_dir();
```

Comment:
As I remember we did not agreed on the form how or if we will support drop in directory. Therefore providing the functionality in api is not a good idea.

Reply:
I still think drop-in directory support is a good idea. Dnf/microdnf is used in multiple distributions. And users are used to using drop-in directories to configure other applications.
I will now make the method private now until we done decision.

Reply:
@j-mracek Methods to load main configuration from files in directory are private now and are added in separate commit. TODO was added.